### DISCUSSION TITLE
Telnyx and Hosted Messaging

### DISCUSSION THREAD

**Discussion by jvoiptel** [0perationPrivacy/VoIP#72] [Ideas]
Hi folks!

So far my testing is successful. One thing I wanted to bring out is Telnyx's cool Hosted Messaging feature. You can have them host SMS without owning the numbers with them. I recently enabled hosted messaging and it's working great. However, I did notice that my VOIP Suite app is not listing hosted SMS numbers. Not sure if this will be something that is needed/wanted in the future. Nice job on recent features! 

See: https://telnyx.com/resources/hosted-messaging-telnyx

**Comment by 0perationPrivacy** [comment]
Thanks, have not tried that feature. Requires LOA and owning a landline, proof of ownership, etc. I won't be able to try it out myself unfortunately. I guess they're intercepting the messages from the other provider with their permission or having them forward it.

Telnyx says you can't view the number in their dashboard but is accessible through their API.
here: https://developers.telnyx.com/docs/api/v2/messaging/Messaging-Hosted-Number

I can look into pulling in that info in the app, but I have no means to test it.
I'll also need to put in exceptions for such numbers to disable calling features as these special case numbers will only support sms/mms.
Will look into it and update here soon.
Thanks.

**Comment by jvoiptel** [comment]
Great, thank you. Actually, it's not SMS forwarding or interception as much as how SMS works with carriers. Any carrier can host SMS on their platform with proper authorization (LOA), if they support the feature of course. It looks like [Twilio ](https://www.twilio.com/docs/phone-numbers/hosted-numbers )supports SMS hosting as well. 

I will gladly test this for you as I have a few hosted SMS DIDs on my Telnyx account. That's actually how I noticed this. Thanks for looking into it.
